src/views/NDAWrapper/index.tsx

```diff
@@ -14,7 +14,10 @@ const NDAWrapper = ({ children }: NDAWrapperProps) => {
 
   useEffect(() => {
     if (user?.acceptedNDA && user?.acceptedNDA?.agreed === false) {
-      history.push('/pre-decisional-notice');
+      history.push({
+        pathname: '/pre-decisional-notice',
+        state: { nextState: pathname !== '/pre-decisional-notice' && pathname }
+      });
     }
```

Comment:
I realize this piece of code doesn't preserve any `search` params.  For this to accurately work with NDA signing, you need to append `search` to `pathname` if the model plan ID is to be parsed in the `Unfollow` component.  Something like this:

`const { pathname, search } = useLocation();`

Then the `nextState` field could contain something like so:
`pathname !== '/pre-decisional-notice' && pathname + (search || '')`